### ISSUE TITLE
Add push notifications on updates to a repository

### ISSUE THREAD

**Issue by KrishnaKumarHariprasannan** [jupyterlab/jupyterlab-git#858]
## Is your feature request related to a problem? Please describe.

In a collaborative environment, where two or more people are working on a shared notebook (or any other document), it will be useful to know when there are updates done on that notebook by others and pull those changes.

## Describe the solution you'd like

Github PR shows a "Refresh" button in PRs and Confluence has a "Reload Page" notification for such scenarios - a similar toast and/or notification on the sidebar can help (not sure this is the right UX). To be clear, we would `fetch`ing for updates (to raise notifications) and `apply` those changes only upon confirmation from the user.

## Describe alternatives you've considered

None

**Comment by fcollonval** [comment]
Hey @KrishnaKumarHariprasannan thank you for reaching out.

There is already an indicator showing (a orange point) on the push/pull icons if you are on the latest commit (see https://github.com/jupyterlab/jupyterlab-git/releases/tag/v0.23.0). Is that what you are looking for? Or would you like this information presented differently?

**Comment by KrishnaKumarHariprasannan** [comment]
Hi @fcollonval, Thanks for your response!

> Is that what you are looking for? Or would you like this information presented differently?

This is definitely closer to the workflow I had in mind - my previous update was meant to suggest adding notifications at a file level (the current indicator shows the changes at a repository level) - from a given set of changes available, one would care more about the files that they're currently working on and so we might want to treat those differently (with higher urgency/importance). Does that sound good to you?

**Comment by fcollonval** [comment]
> Does that sound good to you?

Yes thank you for the clarification. So indeed this will be a new feature as it will require to figure out precisely which files are behind.

**Comment by andrewfulton9** [comment]
@KrishnaKumarHariprasannan how do you envision the file notifications working? Some options I have thought of are 
 1) add a section to the git panel with upstream changed files and then the listed files, 
2) add notification badges to the files in the file browser panel or 
3) have a popup that the user has to acknowledge with the changed files listed. 

What are your thoughts?
The second option here would probably take a fair amount of effort with the others (1, 3) being easier

**Comment by KrishnaKumarHariprasannan** [comment]
@andrewfulton9, a combination of 1. and 3. would work well.

Elaborating on the above, the extension would
* Show indicator on the pull button and list upstream file changes (in a separate panel) when the main/current branch
    * has changes that does not overlap the currently open/modified files
    * has changes that overlaps the currently open/modified files
* Raise toast notifications when the main/current branch
    * has changes that overlaps the currently open/modified files

How does that sound?

**Comment by andrewfulton9** [comment]
@KrishnaKumarHariprasannan, I've been working on this in a [branch on my fork](https://github.com/andrewfulton9/jupyterlab-git/tree/remote_changes_notification).

So far I have:
- added remote changes section
- filled remote changes section with files that have changed on remote branch
- added B status symbol next to file to show that it is behind on local repo. 
![image](https://user-images.githubusercontent.com/13774419/122596993-b64a4a00-d038-11eb-9642-0f214fa71e58.png)

I am wondering on what people's thoughts are for what available options for the files in the remote section should be. Currently I just have it so that a user can open the local version of the file. I was thinking on adding the capability to checkout the remote file into the current branch, raising a warning if there are local changes that would be overwritten, and also to see the diff between the remote file and the local file.

**Comment by fcollonval** [comment]
Hey @andrewfulton9 

Thanks for working on this one. This looks great. The features you are proposing (aka pull only one file and show the diff) would be a great addition too. Don't hesitate to start a draft PR.

**Comment by andrewfulton9** [comment]
Thanks @fcollonval! I added draft PR #962.
